src/context/field.ts

```diff
@@ -0,0 +1,57 @@
+import { invariant } from 'outvariant'
+import { ResponseTransformer } from '../response'
+import { devUtils } from '../utils/internal/devUtils'
+import { jsonParse } from '../utils/internal/jsonParse'
+import { mergeRight } from '../utils/internal/mergeRight'
+import { json } from './json'
+
+type ForbiddenFieldNames = '' | 'data' | 'errors' | 'extensions'
+
+/**
+ * Set a custom field on the GraphQL mocked response.
+ * @example res(ctx.fields('customField', value))
+ * @see {@link https://mswjs.io/docs/api/context/field}
+ */
+export const field = <FieldNameType extends string, FieldValueType>(
+  fieldName: FieldNameType extends ForbiddenFieldNames ? never : FieldNameType,
+  fieldValue: FieldValueType,
+): ResponseTransformer<string> => {
+  return (res) => {
+    validateFieldName(fieldName)
+
+    const prevBody = jsonParse(res.body) || {}
+    const nextBody = mergeRight(prevBody, { [fieldName]: fieldValue })
+
+    return json(nextBody)(res as any) as any
+  }
+}
+
+function validateFieldName(fieldName: string) {
```

Comment:
We should export this function and unit test it. 

Also, let's include an explicit return type:

```suggestion
function validateFieldName(fieldName: string): void {
```